src/Sparrow/ByteString.cs

```diff
@@ -605,8 +661,22 @@ private ByteString Create(void* ptr, int length, int size, ByteStringType type =
 
         private ByteString AllocateWholeSegment(int length, ByteStringType type)
         {
-            // The allocation is big, therefore we will just allocate the segment and move on.                
-            var segment = AllocateSegment(length + sizeof(ByteStringStorage));
+            // The allocation is big, therefore we will just allocate the segment and move on.   
+
+            SegmentInformation segment;
+            if (length > MaxAllocationBlockSizeInBytes)
+            {
+                var memory = new UnmanagedGlobalSegment(length);
+                byte* start = (byte*)memory.Segment.ToPointer();
+                byte* end = start + memory.Size;
+                                
+                segment = new SegmentInformation { Memory = memory, Start = start, Current = start, End = end, CanDispose = true };
+                _wholeSegments.Add(segment);
```

Comment:
In fact, I just realized we can do with far less code. I will change that tomorrow morning.